circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/replica/ReplicaTest.java

```diff
@@ -423,41 +466,106 @@ public void alteringExistingPartitionedReplicaTableWithPartitionsSucceeds() thro
+    }
 
-    verify(mockMetaStoreClient).setPartitionColumnStatistics(setStatsRequestCaptor.capture());
-    SetPartitionsStatsRequest statsRequest = setStatsRequestCaptor.getValue();
+    assertThat(columnStats.size(), is(numTestAlterPartitions + numTestAddPartitions));
 
-    List<ColumnStatistics> columnStats = new ArrayList<>(statsRequest.getColStats());
-    Collections.sort(columnStats, new Comparator<ColumnStatistics>() {
-      @Override
-      public int compare(ColumnStatistics o1, ColumnStatistics o2) {
-        return o1.getStatsDesc().getPartName().compareTo(o2.getStatsDesc().getPartName());
-      }
-    });
-    assertThat(columnStats.size(), is(2));
+    for (int colStatNdx = 0; colStatNdx < numTestAlterPartitions; colStatNdx++) {
+      assertThat(columnStats.get(colStatNdx).getStatsDesc().isIsTblLevel(), is(false));
+      assertThat(columnStats.get(colStatNdx).getStatsDesc().getDbName(), is(DB_NAME));
+      assertThat(columnStats.get(colStatNdx).getStatsDesc().getTableName(), is(TABLE_NAME));
+      assertThat(columnStats.get(colStatNdx).getStatsDesc().getPartName(),
+              is(String.format("c=exist_%s/d=exist_%s_sub", colStatNdx, colStatNdx)));
+      assertThat(columnStats.get(colStatNdx).getStatsObj().size(), is(2));
+    }
 
-    assertThat(columnStats.get(0).getStatsDesc().isIsTblLevel(), is(false));
-    assertThat(columnStats.get(0).getStatsDesc().getDbName(), is(DB_NAME));
-    assertThat(columnStats.get(0).getStatsDesc().getTableName(), is(TABLE_NAME));
-    assertThat(columnStats.get(0).getStatsDesc().getPartName(), is("c=one/d=two"));
-    assertThat(columnStats.get(0).getStatsObj().size(), is(2));
-    assertThat(columnStats.get(1).getStatsDesc().isIsTblLevel(), is(false));
-    assertThat(columnStats.get(1).getStatsDesc().getDbName(), is(DB_NAME));
-    assertThat(columnStats.get(1).getStatsDesc().getTableName(), is(TABLE_NAME));
-    assertThat(columnStats.get(1).getStatsDesc().getPartName(), is("c=three/d=four"));
-    assertThat(columnStats.get(1).getStatsObj().size(), is(2));
+    for (int colStatNdx = numTestAlterPartitions, addPartColStatNdx = 0;
+         colStatNdx < numTestAlterPartitions + numTestAddPartitions;
+         colStatNdx++, addPartColStatNdx++) {
+      assertThat(columnStats.get(colStatNdx).getStatsDesc().isIsTblLevel(), is(false));
+      assertThat(columnStats.get(colStatNdx).getStatsDesc().getDbName(), is(DB_NAME));
+      assertThat(columnStats.get(colStatNdx).getStatsDesc().getTableName(), is(TABLE_NAME));
+      assertThat(columnStats.get(colStatNdx).getStatsDesc().getPartName(),
+              is(String.format("c=new_%s/d=new_%s_sub", addPartColStatNdx, addPartColStatNdx)));
+      assertThat(columnStats.get(colStatNdx).getStatsObj().size(), is(2));
+    }
+  }
+
+
+  @Test
+  public void alteringExistingPartitionedReplicaTableWithNewPartitionsInBatchesSucceeds_0_0() throws TException, IOException {
+    alterExistingPartitionedReplicaTableWithNewPartitionsInBatches(0,0);
```

Comment:
For another day but I wonder if we could use JUnit's "parameterised" feature to do this.

Reply:
I looked into that, but found referenced to Parameterized not working with `MockitoJUnitRunner` (https://stackoverflow.com/questions/12606148/passing-parameterized-input-using-mockitos) so I didn't pursue further in the interest of time.  But yes, that would be the perfect for this situation.

Reply:
Ah, OK. We should move to JUnit5 at some point as then it can definitely be done, see https://github.com/ExpediaGroup/beekeeper/blob/master/beekeeper-path-scheduler-apiary/src/test/java/com/expediagroup/beekeeper/scheduler/filter/WhitelistedListenerEventFilterTest.java